libcloud/loadbalancer/drivers/elb.py

```diff
@@ -308,8 +308,7 @@ def ex_create_balancer_listeners(self, name, listeners=None):
             if protocol == 'HTTPS' or protocol == 'SSL':
                 params['Listeners.member.%d.   \
                         SSLCertificateId' % i] = listener[3]
-        else:
-            return False
+        return False
```

Comment:
@pquentin the suggested change preserves the current code behaviour - since there is no `break` statement in the for loop the `else` block will always be executed - see the [help of the corresponding lgtm code query](https://lgtm.com/rules/5980098/) and [here](http://python-notes.curiousefficiency.org/en/latest/python_concepts/break_else.html) for more details.

personally I do find `for` `else` always a bit confusing as their behaviour is a language dependent convention rather than a logical one - and I'm not very familiar with javascript but I think it has the behaviour you're referring to (else triggered if loop iterable empty)

to return `False` if `listeners` is empty I'd suggest something along the lines of:
```
index = -1
for index, listener in enumerate(listeners):
...
if index == -1:
    return False
```

happy to push that update if that makes sense!